packages/stokhos/CMakeLists.txt

```diff
@@ -72,22 +72,19 @@ IF(Stokhos_ENABLE_Ensemble_Scalar_Type)
   SET(Stokhos_ETI_Default_Ensemble_Size "${Stokhos_ETI_Default_Ensemble_Size_Default}" CACHE STRING "Default ensemble size to use for testing")
   MESSAGE("-- Ensemble size for testing is ${Stokhos_ETI_Default_Ensemble_Size}")
   SET(STOKHOS_DEFAULT_ENSEMBLE_SIZE ${Stokhos_ETI_Default_Ensemble_Size})
-  LIST(FIND Stokhos_ETI_Ensemble_Sizes 4 Stokhos_Ensemble_4_Index)
-  LIST(FIND Stokhos_ETI_Ensemble_Sizes 8 Stokhos_Ensemble_8_Index)
-  LIST(FIND Stokhos_ETI_Ensemble_Sizes 16 Stokhos_Ensemble_16_Index)
-  LIST(FIND Stokhos_ETI_Ensemble_Sizes 32 Stokhos_Ensemble_32_Index)
-  IF(NOT (Stokhos_Ensemble_4_Index EQUAL -1))
-    SET(HAVE_STOKHOS_ENSEMBLE_SIZE_4 1)
-  ENDIF()
-  IF(NOT (Stokhos_Ensemble_8_Index EQUAL -1))
-    SET(HAVE_STOKHOS_ENSEMBLE_SIZE_8 1)
-  ENDIF()
-  IF(NOT (Stokhos_Ensemble_16_Index EQUAL -1))
-    SET(HAVE_STOKHOS_ENSEMBLE_SIZE_16 1)
-  ENDIF()
-  IF(NOT (Stokhos_Ensemble_32_Index EQUAL -1))
-    SET(HAVE_STOKHOS_ENSEMBLE_SIZE_32 1)
-  ENDIF()
+
+  # Make list of enabled ensemble sizes suitable for iteration/searching
+  SET(STOKHOS_MPL_ENSEMBLE_SIZES "typedef Sacado::mpl::vector< ")
+  FOREACH(ENSEMBLE_SIZE ${Stokhos_ETI_Ensemble_Sizes})
+    STRING(APPEND STOKHOS_MPL_ENSEMBLE_SIZES "Sacado::mpl::integral_c<unsigned,${ENSEMBLE_SIZE}>, ")
```

Comment:
Once we have C++14, we can make this an alias to `std::integral_constant`.  Yay for deleting code! :-D